src/pymor/algorithms/timestepping.py

```diff
@@ -10,221 +10,602 @@
+    num_values
+        The number of returned vectors of the solution trajectory. If `None`, each intermediate vector that is
+        calculated is returned. Else an interpolation of the calculated vectors on an equidistant temporal grid is
+        returned, using an appropriate interpolation of the respective time stepper.
     """
 
-    def __init__(self, nt):
+    IteratorType = ExplicitEulerIterator
+
+    def __init__(self, nt, num_values=None, interpolation='P1'):
+        super().__init__(num_values, interpolation)
+
+        assert isinstance(nt, Number)
+        assert nt > 0
+
         self.__auto_init(locals())
 
-    def solve(self, initial_time, end_time, initial_data, operator, rhs=None, mass=None, mu=None, num_values=None):
-        if mass is not None:
-            raise NotImplementedError
-        return explicit_euler(operator, rhs, initial_data, initial_time, end_time, self.nt, mu, num_values)
+
+class ExplicitRungeKuttaIterator(SingleStepTimeStepperIterator):
+
+    def __init__(self, stepper, t0, t1, U0, A, F, M, mu, iter, return_times):
+        super().__init__(stepper, t0, t1, U0, A, F, M, mu, iter, return_times)
+        self.__auto_init(locals())
+        self.dt = dt = (t1 - t0) / stepper.nt
+        # use the ones from base, these are checked and converted in super().__init__()
+        A, F, M, mu = self.A, self.F, self.M, self.mu
+
+        # prepare the function f in d_t y = f(t, y)
+        if not isinstance(M, IdentityOperator) and (not M.parametric or 't' not in M.parameters):
+            M = M.assemble(mu)
+        if not A.parametric or 't' not in A.parameters:
+            A = A.assemble(mu)
+
+        if isinstance(F, ZeroOperator):
+            if isinstance(M, IdentityOperator):
+                def f(t, y):
+                    mu_t = mu.with_(t=t)
+                    return -1*A.apply(y, mu=mu_t)
+            else:
+                def f(t, y):
+                    mu_t = mu.with_(t=t)
+                    return M.apply_inverse(-1*A.apply(y, mu=mu_t), mu=mu_t, initial_guess=y)
+        else:
+            if not F.parametric or 't' not in F.parameters:
+                F = F.assemble(mu)
+            if isinstance(M, IdentityOperator):
+                def f(t, y):
+                    mu_t = mu.with_(t=t)
+                    return F.as_vector(mu=mu_t) - A.apply(y, mu=mu_t)
+            else:
+                def f(t, y):
+                    mu_t = mu.with_(t=t) or Mu({'t': t})
+                    return M.apply_inverse(F.as_vector(mu=mu_t) - A.apply(y, mu=mu_t), mu=mu_t, initial_guess=y)
+
+        self.f = f
+
+    def _step_function(self, U_n, t_n):
+        c, A, b = self.stepper.butcher_array
+        # compute stages
+        s = len(c)
+        stages = U_n.space.empty(reserve=s)
+        for j in range(s):
+            t_n_j = t_n + self.dt*c[j]
+            U_n_j = U_n.copy()
+            for l in range(j):
+                U_n_j.axpy(self.dt*A[j][l], stages[l])
```

Comment:
I think by adding `U_n` as the first element of `stages`, it should be possible to avoid the loop by using `lincomb`?

Reply:
Could be. But since we collect `f(u)` in the stages this would probably require additional code afterwards. So I'd leave it as is, unless you have a concrete proposal...

Reply:
See below.

Reply:
done by now